common/services/prismic/parsers.js

```diff
@@ -544,6 +547,39 @@ export function parseOnThisPage(fragment: PrismicFragment[]): Link[] {
     });
 }
 
+export function parseMediaObjectLinks(
+  fragment: PrismicFragment[]
+): Array<MediaObjectType> {
+  const cannotParse = 'could_not_parse';
+  // filter media_object
+  const filteredMediaObject = fragment.filter(mediaObject => {
+    if (mediaObject.content?.type === 'media_object') {
+      return true;
+    }
+  });
+  // attempt to parse data
+  const parseMediaObjects = filteredMediaObject.map(mediaObject => {
+    try {
+      const { title, text, image } = mediaObject.content.data;
+      return {
+        id: mediaObject.content.id,
+        title: parseTitle(title),
+        text: parseStructuredText(text),
+        image: parseImage(image),
+      };
+    } catch (e) {
+      return cannotParse;
+    }
+  });
+
+  // filter any broken parsers and return this back
+  const filterBadMediaObjects = parseMediaObjects.filter(mediaObject => {
+    return mediaObject !== cannotParse;
+  });
+
+  return filterBadMediaObjects;
```

Comment:
How will editors know they have done something wrong? I am trying to see what the parsing requirements are, and it's not immediately clear?

Reply:
this is basically to catch any issues with the data not coming through , hence  title, text, image. or missing ones. I have a test to make sure.. it doesnt break the page if this is being called.
https://github.com/wellcomecollection/wellcomecollection.org/pull/5650/files#diff-1dffbcdf20c779a943618ded986e2830fb1d9f7e286c8688f95e3394e9f333ba

Reply:
My issue with this is the code is saying: "If the parsers break, just leave out the object". My question is - how do we tell the editors this didn't work?

My preferred method would be to make sure that the parsers work by accounting for optional fields, and just have the component render with those optional fields - this way if something is missing the editors will know.

Reply:
Makes sense. I have done it in a way where if content is missing it should still render on the page what it can render for editors to update etc..